examples/Cloudy/Cloudy_example_ukp.jl

```diff
@@ -0,0 +1,278 @@
+d1 = Parameterized(Normal(4.5, 1.0)) #truth is 5.19
+d2 = Parameterized(Normal(0.0, 2.0)) #truth is 0.378
+d3 = Parameterized(Normal(-1.0, 1.0))#truth is -2.51
+distributions = [d1, d2, d3]
+
+
+prior_mean = [4.5; 0.0; -1.0]
+prior_cov = Array(Diagonal( ones(Float64, length(prior_mean) ) ) )
+
+param_names = ["N0", "u", "k"]
+
+priors = ParameterDistribution(distributions, constraints, param_names)
+
+###
+###  Define the data from which we want to learn the parameters
+###
+
+data_names = ["M0", "M1", "M2"]
+moments = [0.0, 1.0, 2.0]
+n_moments = length(moments)
+
+
+###
+###  Model settings
+###
+
+# Collision-coalescence kernel to be used in Cloudy
+coalescence_coeff = 1/3.14/4/100
+kernel_func = x -> coalescence_coeff
+kernel = Cloudy.KernelTensors.CoalescenceTensor(kernel_func, 0, 100.0)
+
+# Time period over which to run Cloudy
+tspan = (0., 1.0)  
+
+
+###
+###  Generate (artificial) truth samples
+###  Note: The observables y are related to the parameters u by:
+###        y = G(x1, x2) + η
+###
+
+g_settings_true = GModel.GSettings(kernel, dist_true, moments, tspan)
+gt = GModel.run_G(params_true, g_settings_true, PDistributions.update_params, 
+                  PDistributions.moment, Cloudy.Sources.get_int_coalescence)
+
+n_samples = 100
+yt = zeros(length(gt),n_samples)
+# In a perfect model setting, the "observational noise" represent the internal
+# model variability. Since Cloudy is a purely deterministic model, there is no
+# straightforward way of coming up with a covariance structure for this internal
+# model variability. We decide to use a diagonal covariance, with entries
+# (variances) largely proportional to their corresponding data values, gt.
+Γy = convert(Array, Diagonal([100.0, 5.0, 30.0]))
+μ = zeros(length(gt))
+
+# Add noise
+for i in 1:n_samples
+    yt[:, i] = gt .+ rand(MvNormal(μ, Γy))
+end
+
+truth = Observations.Obs(yt, Γy, data_names)
+truth_sample = truth.mean
+
+###
+###  Calibrate: Unscented Kalman Inversion
+###
+
+
+N_iter = 20 # number of iterations
+# need to choose regularization factor α ∈ (0,1],  
+# when you have enough observation data α=1: no regularization
+α_reg =  1.0
+# update_freq 1 : approximate posterior covariance matrix with an uninformative prior
+#             0 : weighted average between posterior covariance matrix with an uninformative prior and prior
+update_freq = 1
+
+process = Unscented(prior_mean, prior_cov, truth_sample, truth.obs_noise_cov, α_reg, update_freq)
+ukiobj = EnsembleKalmanProcessModule.EnsembleKalmanProcess(truth_sample, truth.obs_noise_cov, process)
+
+
+                   
+# Initialize a ParticleDistribution with dummy parameters. The parameters 
+# will then be set in run_G_ensemble
+dummy = 1.0
+dist_type = PDistributions.Gamma(dummy, dummy, dummy)
+g_settings = GModel.GSettings(kernel, dist_type, moments, tspan)
+
+
+
+err = zeros(N_iter) 
+for i in 1:N_iter
+    # prediction step 
+    u_p = EnsembleKalmanProcessModule.update_ensemble_prediction!(ukiobj) 
```

Comment:
`u_p` can be calculated from within the ensemble update step. If you still want to retain the conceptual setup of `analysis` step and `prediction` step. Then I would suggest the following.
Define the function `update_ensemble!` as
```julia
function update_ensemble!(ukiobj::EnsembleKamanProcess{..,Unscented}, g_ens}
    #perform analysis on the model runs
    update_ensemble_analysis!(ukiobj,g_ens)
    #perform new prediction output to model parameters u_p
    return update_ensemble_prediction!(ukiobj.process) 
end
```
Note we don't want `update_ensemble_prediction` to depend on `ukiobj` as we want to use this function in the constructor of the ukiobj. (It can however depend on the process)

Reply:
Updated!